main/src/ca/uwaterloo/flix/language/ast/Scheme.scala

```diff
@@ -114,51 +117,60 @@ object Scheme {
 
   /**
     * Returns `Success` if the given scheme `sc1` is smaller or equal to the given scheme `sc2`.
+    *
+    * Θₚ [T/α₂]π₂ ⊫ₑ {π₁, τ₁ = [T/α₂]τ₂} ⤳! ∙ ; R
+    * T new constructors
+    * ---------------------------------------
+    * Θₚ ⊩ (∀α₁.π₁ ⇒ τ₁) ≤ (∀α₂.π₂ ⇒ τ₂)
     */
-  def checkLessThanEqual(sc1: Scheme, sc2: Scheme, classEnv: Map[Symbol.ClassSym, Ast.ClassContext], eqEnv: ListMap[Symbol.AssocTypeSym, Ast.AssocTypeDef])(implicit flix: Flix): Validation[Substitution, UnificationError] = {
-
-    ///
-    /// Special Case: If `sc1` and `sc2` are syntactically the same then `sc1` must be less than or equal to `sc2`.
-    ///
-    if (sc1 == sc2) {
-      return Validation.success(Substitution.empty)
+  def checkLessThanEqual(sc1: Scheme, sc2: Scheme, cenv0: Map[Symbol.ClassSym, Ast.ClassContext], eenv0: ListMap[Symbol.AssocTypeSym, Ast.AssocTypeDef])(implicit flix: Flix): Validation[Substitution, UnificationError] = {
+
+    // Instantiate sc2, creating [T/α₂]π₂ and [T/α₂]τ₂
+    val (cconstrs2_0, econstrs2_0, tpe2_0) = Scheme.instantiate(sc2, SourceLocation.Unknown)(Level.Default, flix)
+
+    // Resolve what we can from the new econstrs
+    // TODO ASSOC-TYPES probably these should be narrow from the start
+    val tconstrs2_0 = econstrs2_0.map { case Ast.BroadEqualityConstraint(t1, t2) => TypingConstraint.Equality(t1, t2, Provenance.Match(t1, t2, SourceLocation.Unknown)) }
```

Comment:
Easy to add Provenance.Scheme (?).

Also could checkLessThanEqual not take a loc, so we avoid unknown which is always the absolute worst.

Reply:
The locations used here are essentially irrelevant, because are only put into types.

After merge, this function can change to return a Boolean rather than a result, because that's all we use in the new Typer. That means provenance is irrelevant as well.

Reply:
Added to followup